cfme/tests/infrastructure/test_vm_reconfigure.py

```diff
@@ -91,22 +94,24 @@ def ensure_vm_running(full_vm):
         raise Exception("Unknown power state - unable to continue!")
 
 
-@pytest.fixture(params=["cold", "hot"])
-def vm_state(request, full_vm):
-    if request.param == "cold":
-        if full_vm.is_pwr_option_available_in_cfme(full_vm.POWER_OFF):
-            full_vm.mgmt.ensure_state(VmState.STOPPED)
-            full_vm.wait_for_vm_state_change(full_vm.STATE_OFF)
-        else:
-            raise Exception("Unknown power state - unable to continue!")
+def _vm_state(vm, state):
+    if state == "cold" and vm.is_pwr_option_available_in_cfme(vm.POWER_OFF):
+        ensure_state = VmState.STOPPED
+        state_change = vm.STATE_OFF
+    elif state == "hot" and vm.is_pwr_option_available_in_cfme(vm.POWER_ON):
+        ensure_state = VmState.RUNNING
+        state_change = vm.STATE_ON
     else:
-        if full_vm.is_pwr_option_available_in_cfme(full_vm.POWER_ON):
-            full_vm.mgmt.ensure_state(VmState.RUNNING)
-            full_vm.wait_for_vm_state_change(full_vm.STATE_ON)
-        else:
-            raise Exception("Unknown power state - unable to continue!")
+        raise Exception("Unknown power state - unable to continue!")
```

Comment:
Optional: Umn, I would rather used either `AssertionError` or `ValueError` here to be more specific.